File: services/customer-service/src/main/java/com/djesystems/kawa/customer/domain/Customer.java
package com.djesystems.kawa.customer.domain;

import java.time.Instant;
import java.util.UUID;

/**
 * Modèle métier représentant un client KAWA.
 *
 * Cette classe ne dépend volontairement pas de JPA.
 */
public record Customer(
        UUID id,
        String firebaseUid,
        String publicKawaId,
        String email,
        CustomerStatus status,
        Instant createdAt,
        Instant updatedAt
) {
}

File: apps/kawa-platform/src/types/Customer.ts
export type Customer = {
  publicKawaId: string;
  email: string | null;
  status: "ACTIVE" | "SUSPENDED" | "DELETED";
  autoRetailerAssociationEnabled: boolean;
  createdAt: string;
};

export type KawaIdResponse = {
  kawaId: string;
};
